### ISSUE TITLE
Gitlab support?

### ISSUE THREAD

**Issue by jameshwc** [bitsbeats/drone-tree-config#12]
I need to apply drone to a monorepo in gitlab. I'm willing to fork this repo, however, I'm newbie to go. Would you mind giving me some info or hint to help me kick off?

**Comment by foosinn** [comment]
Hey,

you will need to implement a new provider following this interface:
```go
type ScmClient interface {
	ChangedFilesInPullRequest(ctx context.Context, pullRequestID int) ([]string, error)
	ChangedFilesInDiff(ctx context.Context, base string, head string) ([]string, error)
	GetFileContents(ctx context.Context, path string, commitRef string) (
		fileContent string, err error)
	GetFileListing(ctx context.Context, path string, commitRef string) (
		fileListing []FileListingEntry, err error)
}
```

Github example:

https://github.com/bitsbeats/drone-tree-config/blob/master/plugin/scm_clients/github_client.go

**Comment by jezhodges** [comment]
See #19

**Comment by foosinn** [comment]
Closed by #19

**Comment by foosinn** [comment]
Thanks!
